js/spa/test/test_chatinterface_streaming_echo.spec.ts

```diff
@@ -88,3 +88,21 @@ for (const test_case of cases) {
 		);
 	});
 }
+
+test("test stopping generation", async ({ page }) => {
+	const submit_button = page.locator(".submit-button");
+	const textbox = page.getByPlaceholder("Type a message...");
+
+	const long_string = "abc".repeat(1000) + "end";
+
+	await textbox.fill(long_string);
+	await submit_button.click();
+
+	await expect(page.locator(".bot.message").first()).toContainText("abc");
+	const stop_button = page.locator(".stop-button");
+
+	await stop_button.click();
+
+	await expect(page.locator(".bot.message").first()).toContainText("abc");
+	await expect(page.locator(".bot.message").first()).not.toContainText("end");
```

Comment:
I think this test would pass even if the stop button didn't do anything since its unlikely that the output has stopped streaming. You could instead check the value of the textbox, wait a tiny amount of time, check the value again, and assert both values are the same

Reply:
yep makes sense, good call